### PR TITLE
Adds a laserarm implant to roboticist traitor only, Adds new syndicate autosurgeon sprite

### PR DESCRIPTION
**Pull request by improvedname**
## About The Pull Request

This adds an laserarm implant to roboticist traitor items for 10 TC, it comes in a autosurgeon for instant implanting due your own lackability of doing so without needing somebody

The laserarm is functionally a Antique Lasergun inside of your arm.
But it carries all arm implants drawbacks. Like retracting on emp and even exploding depending on the severity of the emp.

The autosurgeon it comes with is NOT multiple use. its only 1 use like the cmo's autosurgeon
Also it comes in a special recolor to scream THIS IS A TRAITOR ITEM so detective can scan it

Also adds a sprite recolor by JustRandomGuy
https://cdn.discordapp.com/attachments/678402836106706955/678405024505659392/New_Piskel_6.png

## Why It's Good For The Game

Let the roboman augment himself and implant a robocop implant into themself its a 20 burn laser gun that recharges slowy with all the drawbacks of regular arm implants, You can easily get more OP stuff like Syndi revolver/Ebow implant if anyone dares to argue about balance.

## Changelog
:cl: Improvedname & JustRandomGuy
add: Adds the syndicate laserarm implant to roboticist traitors
add: Syndicate implants now come in a suspicous autosurgeon
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/autosurgeon.dm [hunk 1]
```diff
@@ -12,6 +12,10 @@
 	var/uses = INFINITE
 	var/starting_organ
 
+/obj/item/autosurgeon/syndicate
+	name = "suspicous autosurgeon"
```
Comment (swindly) [inline code/modules/surgery/organs/autosurgeon.dm]:
```suggestion
	name = "suspicious autosurgeon"
```

**Comment by necromanceranne** [comment]
Uplink weapons are almost always ballistics and not lasers or energy weapons with the exception of the ebow, which is more like a hardlight weapon. This breaks that dichotomy.

Also there are plenty of firearms in the uplink and having a nodrop recharging laser is a big bad mistake. Especially with endless ammunition. The antique laser is special for a reason. It is quite easily one of the strongest guns on the station for it's fast recharge and reliable damage and decent capacity.

**Comment by improvedname** [comment]
dude your arm litteraly explodes if it gets emped a lasergun implant is the least of the worries and shows up on literal all adv scanners its job locked + you can always buy a 2 shot revolver or 10toxin+stam damage ebow lol

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=XBphhwnT-2U

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1786,6 +1786,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/storage/box/hug/reverse_revolver
 	restricted_roles = list("Clown")
 
+/datum/uplink_item/role_restricted/laser_arm
+	name = "Laser arm implant"
+	desc = "A laser arm implant which grants you with a functionally antique lasergun technology inside of your arm, Keep away from emps. Comes with a autosurgeon."
```
Comment (Anonmare) [inline code/modules/uplink/uplink_items.dm]: This reads in a way that's awkward.
"(...) grants you a lasergun, functionally similar to the antique laser, inside of your arm (...)" would read much better.

Comment (improvedname) [reply code/modules/uplink/uplink_items.dm]: sorry i missed grammar class

**Comment by Time-Green** [comment]
You don’t REALLY need an auto-implanter. You can ask your partner or an MD to augment your arm and then you can perform additional surgeries, like laser implant, yourself.

**Comment by improvedname** [comment]
its a traitor item that screams out TRAITOR thats why the autosurgeon

**Comment by itseasytosee** [comment]
I have a few issues with this:
A. The syndicate is all about ballistic guns and laser melees in contrast to nanotrasen's energy guns.
B. This invalidates the steal the caps laser gun objective as it shows that the syndicate already have the tech for it.
C. This has nothing to do with how roboticist plays or the tools it has access to there is no reason why this should be Robo only. (Unlike the modified syringe gun, blast cannon, regent dart gun, clown car, etc.)

**Comment by itseasytosee** [comment]
Also what @Time-Green was saying was that if your arm is already augmented you don't need anyone else to install the obvious traitor item.

**Comment by improvedname** [comment]
> I have a few issues with this:
> A. The syndicate is all about ballistic guns and laser melees in contrast to nanotrasen's energy guns.
> B. This invalidates the steal the caps laser gun objective as it shows that the syndicate already have the tech for it.
> C. This has nothing to do with how roboticist plays or the tools it has access to there is no reason why this should be Robo only. (Unlike the modified syringe gun, blast cannon, regent dart gun, clown car, etc.)
> I like the autosurgeon.

A. Ebow, energy recharging weapon infinite ammo
B. Ok but how they make infinite recharging ebow also it has less ammo then the gun itself.
C. Roboticists are all about augmentions and this is an augmention
D. I don't want to force roboticists to be augmented so they can install a silly arm who litteraly explodes on emping and get your whole budy fucked even more because you augmented yourself and super vulnarable to emps

**Comment by improvedname** [comment]
Tl;dr what you want rather have a 20 burn laser gun or a 2 shot revolver where you can print infinite ammo from a autolathe with way faster reloading time

**Comment by improvedname** [comment]
You can argue its loud but it can be a cap gun or a arcade machine for all you know

**Comment by zxaber** [comment]
I'll probably myself never use this (my TC is already allocated to an emag, binary key, and hidden weapons bay). But if I ever do, I'd prefer it use ballistics. You could totally have it use .45 ammo, since that's available in a hacked autolathe already.

**Comment by AdamElTablawy** [comment]
it's literally a cool laser arm how is that NOT robotics themed ffs

besides, nothing says the syndicate can't use energy weapons at all, they have energy swords and energy bows, an arm mounted energy gun isn't too distinct

**Comment by itseasytosee** [comment]
Why do you keep saying how ineffective your item is compared to others?
Also by themed I mean actual game mechanics as in geneticists use DNA injectors so they get an item that lets them shoot DNA into people. If this did not come with an auto surgeon and you had to get augmented or a buddy to do it for you then it would be themed for sure but what your making it out to be right now is an ineffective stealth weapon that explodes your arm when you get hit by an emp.

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
Comment (bobbahbrown) [reply code/modules/uplink/uplink_items.dm]: This is a genuine problem, please consider changing this.

**Comment by dr-bee** [comment]
Why is this a traitor item and not a security accessible implant?

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -1786,6 +1786,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/storage/box/hug/reverse_revolver
 	restricted_roles = list("Clown")
 
+/datum/uplink_item/role_restricted/laser_arm
+	name = "Laser arm implant"
```
Comment (cacogen) [inline code/modules/uplink/uplink_items.dm]:
```suggestion
	name = "laser arm implant"
```

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
Comment (cacogen) [inline code/modules/uplink/uplink_items.dm]:
```suggestion
	desc = "An implant that grants you a recharging laser gun inside your arm. Weak to EMPs. Comes with a syndicate autosurgeon for immediate self-application."
```

**Comment by PKPenguin321** [comment]
cool to see this being implemented after all these years. neither auto-implanters nor job-locked traitor items existed when i first wrote it

**Comment by improvedname** [comment]
@itseasytosee because im not gonna force a must be augmented for roboticists becausse that gives them even a heavier disadvantage when being emped

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
Comment (ATH1909) [reply code/modules/uplink/uplink_items.dm]: it's clearly a reference to the poor english translations of some of the mega man games, smh my head my head

**Comment by necromanceranne** [comment]
I'll be entirely honest, in addition to my previous concerns about it breaking dichotomy, it seems unnecessary to me to even add a self-charging laser gun when as you already stated, there is equipment in the uplink that serves a similar purpose. Even if you COULD buy an ebow (which is not the same as a laser gun in function or practical use anyway even if it is an endlessly charging weapon) or a revolver (a massive investment that isn't as easily utilized for mass murder as this implant would be due to losing speedloader spam), these items do not justify this addition. That actually just admitting the niche is already filled, but this one is technically better than those two due to the no-drop and the self-charge. And as already stated, having something comparable with a traitor objective out of the uplink defeats the purpose of that item being a traitor objective and the unique nature of that item. (Ignore AEGs, god those things suck)

That said, adding cybernetic combat stuff into the uplink for roboticists is an interesting idea, but it shouldn't be just a generic laser gun.

**Comment by improvedname** [comment]
EMP fucks up your day bro. Dont forget hos gun comes with an emp function. Security gets a ion rifle. This thing explodes or refuses to function if it gets emped and can halt whatever your doing because nature its a implant also an energy based implant. You lose charge and have a chance of the implant retracting/exploding when emped

A 357. Cant be emped neither can a ebow

**Comment by improvedname** [comment]
After all the all normal drawbacks of energy weapons apply which also gives more reason to use the ablative trenchcoat

**Comment by CRITAWAKETS** [comment]
if something is meant to be useless why even add it

**Comment by Tlaltecuhtli** [comment]
give this trash to sec and give traitor minigun arm

**Comment by daklj** [comment]
good PR
we need more antag content

**Comment by improvedname** [comment]
Because its not useless @CRITAWAKETS

**Comment by necromanceranne** [comment]
> EMP fucks up your day bro. Dont forget hos gun comes with an emp function. Security gets a ion rifle. This thing explodes or refuses to function if it gets emped and can halt whatever your doing because nature its a implant also an energy based implant. You lose charge and have a chance of the implant retracting/exploding when emped
> 
> A 357. Cant be emped neither can a ebow

I think you're addressing too heavily whether or not it's counterable or if it's balanced. That's not really my major concern over whether it even seems worth adding when it's niche is so readily handled except for the fact that it has no-drop. There are obviously plenty of powerful tools security can employ against it. But that is also not the concern. Thematics (it's thematic for a roboticist, not for outsourced syndicate equipment), functional place in the uplink and too much parity with special crew gear is what I'm concerned about, and you haven't alleviated this in any way. 

One interesting thing about science department syndicate gear is that a lot of it; the hidden weapon compartment, the the Australicus serum injector, the modified syringe gun and the pipe gun, require you working in your department to make the most of it via way of using the resources available from the department in ways that can't be employed otherwise. For something like this implant, it cuts out that element of the department work and goes straight to a powerful murder tool that seems very readily able to be destroyed, by your own admission. 

Rather than add something already seen and not particularly creative, you could use this as an opportunity to add something unique to the roboticist involving the part of the department that hasn't received attention for traitor gear, but still requires some effort on your part to make the most of it. Hidden weapon compartment needs you to craft the mech, borg subversion requires getting access to an upload, brainwashing requires you to be able to kidnap victims for your objective implantation. An implant is a good starting point, but this is mostly a humble request to re-evaluate the idea.

**Comment by improvedname** [comment]
Borg subversion doesn't require access to a upload just a board.

**Comment by ExcessiveUseOfCobblestone** [comment]
We killed the NT energy / Syndie Ballistics lore with steeleweapons tbf.

**Comment by zxaber** [comment]
Does that mean I can finally add 1911s to cargo?

**Comment by ExcessiveUseOfCobblestone** [comment]
weird way to say remove steelerifles (and shotguns mwahahahaha)

**Comment by improvedname** [comment]
Yes but you gotta have somebody else implant the second one

**Comment by AdamElTablawy** [comment]
Fine, just make it a cybernetic ballistic weapon, like an arm-mounted gauss rifle or something. It's not like this is a particularly GOOD item, just a themed one, as it stands-a captain's laser gun with a nodrop implant, but for TC at roundstart.
